### DISCUSSION TITLE
Tox plugins and package adaptation

### DISCUSSION THREAD

**Discussion by brunolnetto** [tox-dev/tox#2533] [Q&A] [answered]
Hi,

I try to use the tox plugin to improve https://pypi.org/project/eule/
. In this case, I reproduce the commands and instructions available on [this website](https://tox.wiki/en/latest/plugins.html). I can run `python setup.py sdist bdist_wheel` and publish with the command `twine upload --skip-existing dist/*` if I wish to publish it. 

However, I cannot adequately format the original project correctly. Would you provide some further instructions? Please, do not hold back pointint presumably clear instructions out!

**Comment by gaborbernat** [comment, marked as answer]
It's not clear to me what issues you're having so can't really help. 

> I can run `python setup.py sdist bdist_wheel` 

note you should not be doing this per https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html

**Comment by jugmac00** [comment]
@brunolnetto Are you looking for a way to use a formatter for your project? ie black via tox? https://github.com/psf/black

I gave a lightning talk on that topic at PyOhio:
https://www.youtube.com/watch?v=OnM3KuE7MQM

**Comment by brunolnetto** [comment]
It is rather a (mis)understanding of mine than an issue. I am able to create the tox-BLA as it is on instructions. My attempt is to refactor my tox.ini, further configurations files, and project structure to manually inherit these tox good practices.

**Comment by brunolnetto** [comment]
Imagine I am a cooker and can prepare a good soup. But I read from this recipe book the same recipe, but with better-prepared ingredients and manners. I want to set up my own recipe with these good preparation steps.

**Comment by brunolnetto** [comment]
Hallo @jugmac00 , 

Allererstes bedanke ich bei Dir für die Bemühung. Ich versuche eine sehr gute Paketbeispiel zur Verfügung stehen, damit künftige Entwickler sich darauf basieren können. Falls du davon ausprobieren möchtest, kanns du selbst es mit Befehl `pip install eule` installieren. 

Atschüss.

**Reply by jugmac00** [reply]
While I appreciate that you try to be polite and write in my native language, it is an issue both with the other maintainers and the other readers. Please don't do that.

**Reply by brunolnetto** [reply]
Ok!

**Comment by jugmac00** [comment]
Hi Bruno,

I had a look at your repository. It looks pretty decent. I noticed a couple of issues, and fixed at least a couple ones so you can run tox.

This had nothing to do with tox per se, it was an issue with packaging - so also a `pip install .` would have failed.

see
https://github.com/quivero/eule/pull/8

Here we support tox, and not how a perfect project setup should look like - there are many tutorials out there.

Cheers,
Jürgen

**Reply by brunolnetto** [reply]
"It looks pretty decent." is the best english translation to "Nicht schlecht".

**Comment by brunolnetto** [comment]
@jugmac00 Look out grown-up library doing some grown-up stuff. They run so fast these days...

https://github.com/soup-engredient/experiments/tree/master/pizzayolo
